### ISSUE TITLE
ProjectProvider & ProjectFactory -> VFPT.Core

### ISSUE THREAD

**Issue by cloudRoutine** [fsprojects/VisualFSharpPowerTools#1411]
_Dependent upon completion of Open Documents Tracker_ #1410 

`IProjectProvider` interface will be defined and the `ProjectProvider` type will be implemented in VFPT.Core. (Logic may still need to have a `VsProjectProvider` that implements the `IProjectProvider` interface). The `ProjectProvider` implementation can probably be changed to acquire compilation flags, project contents and configurations settings, 

_WIP - Need to review further to layout how to split roles between `ProjectFactory` in Core and Logic_

Code Located @ 
https://github.com/fsprojects/VisualFSharpPowerTools/blob/master/src/FSharpVSPowerTools.Logic/ProjectSystem/ProjectProvider.fs
https://github.com/fsprojects/VisualFSharpPowerTools/blob/master/src/FSharpVSPowerTools.Logic/ProjectSystem/ProjectFactory.fs

**Comment by vasily-kirichenko** [comment]
I'd like to rename it to `ISolution`/`Solution`.

**Comment by smoothdeveloper** [comment]
@cloudRoutine having this moved to core would simplify work I'm doing on GoToDefinitionFilter as i'd need not to get rid of it / `ISolution`.

So for now, I won't move stuff to Core but focus on other remaining dependencies.

**Comment by smoothdeveloper** [comment]
@vasily-kirichenko I agree we should have something for `Solution` but we can't just rename it I believe, for example `GetReferencedProjects` doesn't make sense on `Solution` and probably other inconsistencies.

I think we can start define `Solution` and move the "identified once for all projects in solution" aspects to that interface when they become identified.

What do you all think?

**Comment by vasily-kirichenko** [comment]
Solution concept for us is just "set of projects", nothing more. I don't think it should do anything beyond simple operations like "opened", "closed", "project loaded", "project updated", etc.

**Comment by smoothdeveloper** [comment]
We can split out ProjectDescriptor (a record with the same member as the get properties in current IProjectProvider) as I was (kind of) describing in #1440, I think this should be done disregarding the fact that #1410 is done or not.
